<?xml version="1.0" encoding="UTF-8"?>
<launch>
        <!-- 0: navigate_through_pose, 1: follow_waypoints -->
        <arg name="waypoint_marker_scale" default="0.5" />
        <arg name="waypoint_marker_color_r" default="1.0" />
        <arg name="waypoint_marker_color_g" default="0.0" />
        <arg name="waypoint_marker_color_b" default="1.0" />
        <arg name="waypoint_marker_color_a" default="1.0" />
        <arg name="waypoint_stop_marker_color_r" default="1.0" />
        <arg name="waypoint_stop_marker_color_g" default="0.0" />
        <arg name="waypoint_stop_marker_color_b" default="0.0" />
        <arg name="waypoint_stop_marker_color_a" default="1.0" />
        <arg name="waypoint_text_marker_scale" default="0.75" />
        <arg name="waypoint_text_marker_color_r" default="1.0" />
        <arg name="waypoint_text_marker_color_g" default="1.0" />
        <arg name="waypoint_text_marker_color_b" default="1.0" />
        <arg name="waypoint_text_marker_color_a" default="1.0" />

        <!-- map loader -->
        <node name="nav2_waypoint_publisher" pkg="nav2_waypoint_publisher" exec="nav2_waypoint_publisher" output="screen">
                <param name="waypoint_marker_scale" value="$(var waypoint_marker_scale)" />
                <param name="waypoint_marker_color_r" value="$(var waypoint_marker_color_r)" />
                <param name="waypoint_marker_color_g" value="$(var waypoint_marker_color_g)" />
                <param name="waypoint_marker_color_b" value="$(var waypoint_marker_color_b)" />
                <param name="waypoint_marker_color_a" value="$(var waypoint_marker_color_a)" />
                <param name="waypoint_stop_marker_color_r" value="$(var waypoint_stop_marker_color_r)" />
                <param name="waypoint_stop_marker_color_g" value="$(var waypoint_stop_marker_color_g)" />
                <param name="waypoint_stop_marker_color_b" value="$(var waypoint_stop_marker_color_b)" />
                <param name="waypoint_stop_marker_color_a" value="$(var waypoint_stop_marker_color_a)" />
                <param name="waypoint_text_marker_scale" value="$(var waypoint_text_marker_scale)" />
                <param name="waypoint_text_marker_color_r" value="$(var waypoint_text_marker_color_r)" />
                <param name="waypoint_text_marker_color_g" value="$(var waypoint_text_marker_color_g)" />
                <param name="waypoint_text_marker_color_b" value="$(var waypoint_text_marker_color_b)" />
                <param name="waypoint_text_marker_color_a" value="$(var waypoint_text_marker_color_a)" />
        </node>
</launch>
